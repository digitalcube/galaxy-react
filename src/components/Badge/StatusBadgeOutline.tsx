import React, { FC, PropsWithChildren } from 'react';
import { memo } from 'react';
import { useMemo } from 'react';
import { StatusLabelName } from '../../interfaces';
import { FaCircle } from 'react-icons/fa';

export type StatusBadgeOutlineProps = PropsWithChildren<{
  id?: string;
  status?: StatusLabelName | '';
  className?: string;
  labelPrefix?: string;
	labelSuffix?: string;
	withExternalLinkIcon?: boolean;
}>;
export const StatusBadgeOutline: FC<StatusBadgeOutlineProps> = memo(
  ({ id, status, className, labelPrefix, labelSuffix, withExternalLinkIcon, children }) => {
    const statusClassName = useMemo(() => {
      switch (status) {
        case 'info':
          return 'c-ready';
        case 'warning':
          return 'c-is-pending';
        case 'danger':
          return 'c-is-failed';
        case 'success':
          return 'c-is-enabled';
        case 'stopped':
        case 'running':
        case 'starting':
        case 'starting-static':
          return `c-is-site-${status}`;
        case 'attached':
        case 'deployed':
        case 'enabled':
        case 'failed':
        case 'verified':
        case 'pending':
        case 'disabled':
          return `c-is-${status}`;
        case 'loading':
          return 'c-is-pending';
        case 'updating':
        case 'provisioning':
        case 'inprogress':
          return 'c-is-site-starting';
        case 'generating':
        case 'building':
          return 'c-building';
        case 'published':
        case 'onteam':
        case 'scheduled':
        case 'ready':
        default:
          return `c-${status}`;
      }
    }, [status]);
    const classNames = [
      'rounded font-weight-bold c-site-status c-site-status-outline',
      statusClassName,
      className,
    ].filter(Boolean);
    const label = useMemo(() => {
      if (!status) return '';
      return [
        labelPrefix ? `${labelPrefix} ` : '',
        status.charAt(0).toUpperCase() + status.slice(1),
        labelSuffix ? ` ${labelSuffix}` : '',
      ].filter(Boolean);
		}, [status, labelPrefix, labelSuffix]);

    if (!status) return null;
    return (
			<div id={id} className={withExternalLinkIcon ? classNames.join(' ')+" site-status-outline-icon" : classNames.join(' ')} >
				{withExternalLinkIcon ? (
        <i className="ml-2" aria-hidden="true">
          <FaCircle />
        </i>
      ) : null}
        {children || label}
      </div>
    );
  }
);
