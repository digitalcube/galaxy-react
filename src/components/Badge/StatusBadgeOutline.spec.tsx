import { render } from '@testing-library/react';
import React from 'react';
import { StatusBadgeOutline } from './StatusBadgeOutline';

describe('StatusBadgeOutline', () => {
  describe('snapshot testing per status', () => {
    it.each([
      'stopped' as const,
      'disabled' as const,
      'running' as const,
      'attached' as const,
      'deployed' as const,
      'enabled' as const,
      'pending' as const,
      'starting' as const,
      'building' as const,
      'starting-static' as const,
      'failed' as const,
      'verified' as const,
      'published' as const,
      'onteam' as const,
      'scheduled' as const,
      'ready' as const,
    ])('status is %p', (status) => {
      const { asFragment } = render(<StatusBadgeOutline status={status} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
  describe('prefix', () => {
    it.each([undefined, 'WordPress', 'Domain'])('prefix is %p', (prefix) => {
      const { asFragment } = render(
        <StatusBadgeOutline status="ready" labelPrefix={prefix} />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
  describe('suffix', () => {
    it.each([undefined, 'WordPress', 'Domain'])('suffix is %p', (suffix) => {
      const { asFragment } = render(
        <StatusBadgeOutline status="ready" labelSuffix={suffix} />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
  describe('AddingIcon', () => {
    it('When the icon is attached', () => {
      const { asFragment } = render(
        <StatusBadgeOutline
          status="running"
          labelSuffix="AddingIcon"
          withExternalLinkIcon={true}
        />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
