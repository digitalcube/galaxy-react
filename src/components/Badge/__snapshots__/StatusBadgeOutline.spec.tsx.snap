// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBadgeOutline AddingIcon When the icon is attached 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-is-site-running site-status-outline-icon"
  >
    <i
      aria-hidden="true"
      class="ml-2"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
        />
      </svg>
    </i>
    Running AddingIcon
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline prefix prefix is "Domain" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-ready"
  >
    Domain Ready
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline prefix prefix is "WordPress" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-ready"
  >
    WordPress Ready
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline prefix prefix is undefined 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-ready"
  >
    Ready
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline snapshot testing per status status is "attached" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-is-attached"
  >
    Attached
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline snapshot testing per status status is "building" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-building"
  >
    Building
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline snapshot testing per status status is "deployed" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-is-deployed"
  >
    Deployed
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline snapshot testing per status status is "disabled" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-is-disabled"
  >
    Disabled
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline snapshot testing per status status is "enabled" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-is-enabled"
  >
    Enabled
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline snapshot testing per status status is "failed" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-is-failed"
  >
    Failed
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline snapshot testing per status status is "onteam" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-onteam"
  >
    Onteam
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline snapshot testing per status status is "pending" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-is-pending"
  >
    Pending
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline snapshot testing per status status is "published" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-published"
  >
    Published
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline snapshot testing per status status is "ready" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-ready"
  >
    Ready
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline snapshot testing per status status is "running" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-is-site-running"
  >
    Running
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline snapshot testing per status status is "scheduled" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-scheduled"
  >
    Scheduled
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline snapshot testing per status status is "starting" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-is-site-starting"
  >
    Starting
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline snapshot testing per status status is "starting-static" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-is-site-starting-static"
  >
    Starting-static
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline snapshot testing per status status is "stopped" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-is-site-stopped"
  >
    Stopped
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline snapshot testing per status status is "verified" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-is-verified"
  >
    Verified
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline suffix suffix is "Domain" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-ready"
  >
    Ready Domain
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline suffix suffix is "WordPress" 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-ready"
  >
    Ready WordPress
  </div>
</DocumentFragment>
`;

exports[`StatusBadgeOutline suffix suffix is undefined 1`] = `
<DocumentFragment>
  <div
    class="rounded font-weight-bold c-site-status c-site-status-outline c-ready"
  >
    Ready
  </div>
</DocumentFragment>
`;
